### PR TITLE
RHDEVDOCS-4542 Release Notes for Pipelines 1.8.1

### PR DESCRIPTION
**Pull request by sounix000**
- **Aligned team**: Dev Tools
- **OCP version for cherry-picking**: `enterprise-4.11`, `enterprise-4.12`
- **JIRA issues**: [RHDEVDOCS-4542 Release Notes for Pipelines 1.8.1](https://issues.redhat.com/browse/RHDEVDOCS-4542)
- **Preview pages**: [1.8.1 RN](http://file.pnq.redhat.com/sosarkar/4542-pipelines-1-8-1-RN/cicd/pipelines/op-release-notes.html#release-notes-1-8-1_op-release-notes)
- **SME review**: @rupalibehera 
- **QE review**: @VeereshAradhya

### CODE REVIEW

**Review by rupalibehera — APPROVED** [review]

File: modules/op-release-notes-1-8.adoc [hunk 1]
```diff
@@ -533,3 +533,69 @@ securityContext:
 * Before this update, pods for the prune cronjobs were not scheduled on infrastructure nodes, as expected. Instead, they were scheduled on worker nodes or not scheduled at all. With this update, these types of pods can now be scheduled on infrastructure nodes if configured in the `TektonConfig` custom resource (CR).
 // link:https://issues.redhat.com/browse/SRVKP-1806
 // Rupali Behera
+
+
+[id="release-notes-1-8-1_{context}"]
+== Release notes for {pipelines-title} General Availability 1.8.1
+
+With this update, {pipelines-title} General Availability (GA) 1.7.2 is available on {product-title} 4.9, 4.10, and the upcoming version.
```
Comment (VeereshAradhya) [inline modules/op-release-notes-1-8.adoc]: 1.8.1 on 4.10 and 4.11

File: modules/op-release-notes-1-8.adoc [hunk 2]
```diff
@@ -533,3 +533,69 @@ securityContext:
...
+* Before this update, the pruner deleted task runs of running pipelines and displayed the following warning: `some tasks were indicated completed without ancestors being done`. With this update, the pruner retains the task runs that are part of running pipelines.
+// https://issues.redhat.com/browse/SRVKP-2419
+
+* Before this update, `pipeline-1.8` was the default channel for installing the {pipelines-title} Operator 1.8.x. With this update, `latest` is the default channel.
+// https://issues.redhat.com/browse/SRVKP-2412
+
+* Before this update, the {pac} controller pods did not have access to certificates exposed by the user. With this update, {pac} can now access routes and Git repositories behind a self-signed or a custom certificate.
+// https://issues.redhat.com/browse/SRVKP-2470
+
+* Before this update, the task failed with RBAC errors after upgrading from {[pipelines-title]} 1.7.2 to 1.8.0. With this update, the tasks run successfully without any RBAC errors.
```
Comment (VeereshAradhya) [inline modules/op-release-notes-1-8.adoc:591]: `{[pipelines-title]}` instead of `{pipelines-title}`

File: modules/op-release-notes-1-8.adoc [hunk 3]
```diff
@@ -533,3 +533,69 @@ securityContext:
...
+// https://issues.redhat.com/browse/SRVKP-2472
+
+* Before this update, using the `tkn` CLI tool cluster administrators could not remove the task runs and pipline runs that contained a `result` object of `array` type. With this update, cluster adminstrators can use the `tkn` CLI tool to remove the task runs and pipeline runs that contain a `result` object of `array` type.
```
Comment (VeereshAradhya) [inline modules/op-release-notes-1-8.adoc:594]: This is the case for the normal user also and not only with the administrator right?
@concaf @rupalibehera

Comment (rupalibehera) [reply modules/op-release-notes-1-8.adoc:594]: Yes @VeereshAradhya

**Review by vdemeester — APPROVED** [review]

File: modules/op-release-notes-1-8.adoc [hunk 4]
```diff
@@ -533,3 +533,69 @@ securityContext:
...
+// https://issues.redhat.com/browse/SRVKP-2470
+
+* Before this update, the task failed with RBAC errors after upgrading from {pipelines-title} 1.7.2 to 1.8.0. With this update, the tasks run successfully without any RBAC errors.
+// https://issues.redhat.com/browse/SRVKP-2472
+
+* Before this update, using the `tkn` CLI tool you could not remove the task runs and pipline runs that contained a `result` object of `array` type. With this update, you can use the `tkn` CLI tool to remove the task runs and pipeline runs that contain a `result` object of `array` type.
+// https://issues.redhat.com/browse/SRVKP-2478
+
+* Before this update, if a pipeline specification contained a task with an `ENV_VARS` parameter of `array` type, the pipeline run failed with the following error: `invalid input params for task func-buildpacks: param types don't match the user-specified type: [ENV_VARS]`. With this update, pipeline runs with such pipeline and task specifications run without failure.
```
Comment (gabriel-rh) [inline modules/op-release-notes-1-8.adoc:597]: I found this hard to parse when reading:

"pipeline runs with such pipeline and task specifications run without failure."

The second "run" threw me - maybe say "no longer fail" or "do not fail" to simplify it

File: modules/op-release-notes-1-8.adoc [hunk 5]
```diff
@@ -533,3 +533,69 @@ securityContext:
...
+// https://issues.redhat.com/browse/SRVKP-2422
+
+* Before this update, cluster adminstrators could not provide a `config.json` file to the `Buildah` cluster task for accessing a container registry. With this update, cluster adminstrators can provide the `Buildah` cluster task a `config.json` file using the `dockerconfig` workspace.
```
Comment (gabriel-rh) [inline modules/op-release-notes-1-8.adoc:600]:
```suggestion
* Before this update, cluster adminstrators could not provide a `config.json` file to the `Buildah` cluster task for accessing a container registry. With this update, cluster adminstrators can provide the `Buildah` cluster task with a `config.json` file using the `dockerconfig` workspace.
```

File: modules/op-release-notes-1-8.adoc [hunk 6]
```diff
@@ -533,3 +533,69 @@ securityContext:
...
+[id="fixed-issues-1-8-1_{context}"]
+=== Fixed issues
+
+* Before this update, the pruner deleted task runs of running pipelines and displayed the following warning: `some tasks were indicated completed without ancestors being done`. With this update, the pruner retains the task runs that are part of running pipelines.
+// https://issues.redhat.com/browse/SRVKP-2419
+
+* Before this update, `pipeline-1.8` was the default channel for installing the {pipelines-title} Operator 1.8.x. With this update, `latest` is the default channel.
+// https://issues.redhat.com/browse/SRVKP-2412
+
+* Before this update, the {pac} controller pods did not have access to certificates exposed by the user. With this update, {pac} can now access routes and Git repositories behind a self-signed or a custom certificate.
```
Comment (gabriel-rh) [inline modules/op-release-notes-1-8.adoc:588]: "behind a cert" doesn't seem to be a very common phrase (although it does have some hits on google) - maybe re-word?

**Review by gabriel-rh — COMMENTED** [review]
Looks good - just some minor suggestions

File: modules/op-release-notes-1-8.adoc [hunk 7]
```diff
@@ -533,3 +533,69 @@ securityContext:
 * Before this update, pods for the prune cronjobs were not scheduled on infrastructure nodes, as expected. Instead, they were scheduled on worker nodes or not scheduled at all. With this update, these types of pods can now be scheduled on infrastructure nodes if configured in the `TektonConfig` custom resource (CR).
 // link:https://issues.redhat.com/browse/SRVKP-1806
 // Rupali Behera
+
+
+[id="release-notes-1-8-1_{context}"]
+== Release notes for {pipelines-title} General Availability 1.8.1
+
+With this update, {pipelines-title} General Availability (GA) 1.8.1 is available on {product-title} 4.10, 4.11, and the upcoming version.
```
[the remark this answers is not in the document]
Comment (sounix000) [reply modules/op-release-notes-1-8.adoc:541]: Marking the comment as resolved. But as you said - it stays just in case... This is the first comment, so it shouldn't be very hard to find out.

File: modules/op-release-notes-1-8.adoc [hunk 8]
```diff
@@ -533,3 +533,69 @@ securityContext:
...
+
+[id="known-issues-1-8-1_{context}"]
+=== Known issues
+
+* When installer sets associated with cluster tasks are in failed state, the status of the `TektonConfig` custom resource is incorrectly displayed as `True` instead of `False`.
```
Comment (VeereshAradhya) [inline modules/op-release-notes-1-8.adoc:546]: @sm43 Is this only related with clustertasks related installerset or with any installersets in general?

Comment (sm43) [reply modules/op-release-notes-1-8.adoc:546]: in general, if there is any error in addon due to some reason
it was not getting reflected in tektonconfig
but this will be part of 1.9 not 1.8.1

Comment (sm43) [reply modules/op-release-notes-1-8.adoc:546]: for 1.8.1 the installer set was failing due to missing clustertask field, that bug is fixed.

Comment (VeereshAradhya) [reply modules/op-release-notes-1-8.adoc:546]: This is part of known issues. @sounix000 this is not only related to clustertask related installersets, if any installerset is in error state the config is shown as `True`

**Review by gabriel-rh — APPROVED** [review]
/lgtm

File: modules/op-release-notes-1-8.adoc (hunk 7 above)
[the remark this answers is not in the document]
Comment (sounix000) [reply modules/op-release-notes-1-8.adoc:541]: On a second thought, removed the "upcoming version". QE also suggested similarly.

**Review by VeereshAradhya — APPROVED** [review]

File: modules/op-release-notes-1-8.adoc [hunk 9]
```diff
@@ -533,3 +533,69 @@ securityContext:
...
+// https://issues.redhat.com/browse/SRVKP-2472
+
+* Before this update, using the `tkn` CLI tool you could not remove the task runs and pipline runs containing a `result` object of `array` type. With this update, you can use the `tkn` CLI tool to remove the task runs and pipeline runs containing a `result` object of `array` type.
+// https://issues.redhat.com/browse/SRVKP-2478
+
+* Before this update, if a pipeline specification contained a task with an `ENV_VARS` parameter of `array` type, the pipeline run failed with the following error: `invalid input params for task func-buildpacks: param types don't match the user-specified type: [ENV_VARS]`. With this update, pipeline runs with such pipeline and task specifications do not fail.
+// https://issues.redhat.com/browse/SRVKP-2422
+
+* Before this update, cluster adminstrators could not provide a `config.json` file to the `Buildah` cluster task for accessing a container registry. With this update, cluster adminstrators can provide the `Buildah` cluster task with a `config.json` file using the `dockerconfig` workspace.
```
Comment (rolfedh) [inline modules/op-release-notes-1-8.adoc:600]:
```suggestion
* Before this update, cluster administrators could not provide a `config.json` file to the `Buildah` cluster task for accessing a container registry. With this update, cluster administrators can provide the `Buildah` cluster task with a `config.json` file by using the `dockerconfig` workspace.
```

**Review by rolfedh — COMMENTED** [review]
Hi @sounix000 and @gabriel-rh. Nice work, overall. For our future PRs, I wonder how we might catch spelling, punctuation, and grammar issues like these earlier? This way, they won't come up and delay possibly time-sensitive merge requests. What do you think?

File: modules/op-release-notes-1-8.adoc [hunk 10]
```diff
@@ -533,3 +533,69 @@ securityContext:
...
+
+[id="release-notes-1-8-1_{context}"]
+== Release notes for {pipelines-title} General Availability 1.8.1
+
+With this update, {pipelines-title} General Availability (GA) 1.8.1 is available on {product-title} 4.10 and 4.11.
+
+[id="known-issues-1-8-1_{context}"]
+=== Known issues
+
+* When installer sets are in failed state, the status of the `TektonConfig` custom resource is incorrectly displayed as `True` instead of `False`.
```
[the remark this answers is not in the document]
Comment (gabriel-rh) [reply modules/op-release-notes-1-8.adoc]: I didn't really have a problem with this - if it is an explicit state like "stopped state" or "paused state", I would probably use the definite article "the stopped state" rather the "a stopped state" but that is just my opinion.

File: modules/op-release-notes-1-8.adoc [hunk 11]
```diff
@@ -533,3 +533,69 @@ securityContext:
...
+* Before this update, `pipeline-1.8` was the default channel for installing the {pipelines-title} Operator 1.8.x. With this update, `latest` is the default channel.
+// https://issues.redhat.com/browse/SRVKP-2412
+
+* Before this update, the {pac} controller pods did not have access to certificates exposed by the user. With this update, {pac} can now access routes and Git repositories guarded by a self-signed or a custom certificate.
+// https://issues.redhat.com/browse/SRVKP-2470
+
+* Before this update, the task failed with RBAC errors after upgrading from {pipelines-title} 1.7.2 to 1.8.0. With this update, the tasks run successfully without any RBAC errors.
+// https://issues.redhat.com/browse/SRVKP-2472
+
+* Before this update, using the `tkn` CLI tool you could not remove the task runs and pipline runs containing a `result` object of `array` type. With this update, you can use the `tkn` CLI tool to remove the task runs and pipeline runs containing a `result` object of `array` type.
```
[the remark this answers is not in the document]
Comment (gabriel-rh) [reply modules/op-release-notes-1-8.adoc]: As someone who probably over-uses commas, I don't have a problem leaving out the comma in "using the `tkn` CLI tool you could not" as I think it helps the flow when reading.

I don't have a problem with "the task runs... containing" - using the past tense seems more stilted to me

I was tempted to suggest "object of type `array`" as that would be pretty common usage in developer-specific content, but I thought it might be nit-picking.

I know there is a preference for past tense in RNs, but "that contained....whose type was..."   followed by "that contain....whose type is"  just feels a lot more difficult to parse that having parallel "containing....of type" in both places. 

Yes, I did miss pipline - I will copy all content into an editor in future to spell-check - lesson learned.

File: modules/op-release-notes-1-8.adoc (hunk 9 above)
Comment (gabriel-rh) [reply modules/op-release-notes-1-8.adoc]: Fair comment - I did not copy this content into an editor to spell-check and vale - lesson learned.

File: modules/op-release-notes-1-8.adoc (hunk 10 above)
[the remark this answers is not in the document]
Comment (rolfedh) [reply modules/op-release-notes-1-8.adoc]: Hi, Gabriel. In the _IBM Style Guide_, the "Articles" section provides some guidance and examples on this topic.

File: modules/op-release-notes-1-8.adoc (hunk 11 above)
[the remark this answers is not in the document]
Comment (rolfedh) [reply modules/op-release-notes-1-8.adoc]: While I was commenting on a spelling issue ("piplines") and a missing comma at the end of a parenthetical phrase ("tool you"), I included some style preferences that I believe improve clarity (all the other suggestions). You're welcome to take or leave what you want.
